web-registry/src/pages/Roles.tsx

```diff
@@ -86,37 +103,53 @@ const Roles: React.FC = () => {
   const { user } = useAuth0();
   const userEmail = user?.email;
 
+  const { wallet } = useWallet();
+
   const [updateProject] = useUpdateProjectByIdMutation();
   const { data } = useProjectByIdQuery({
     variables: { id: projectId },
     fetchPolicy: 'cache-and-network',
   });
+  const project = data?.projectById;
+  const creditClassId = project?.creditClassByCreditClassId?.onChainId;
+
   const { data: userProfileData } = useGetOrganizationProfileByEmailQuery({
     skip: !userEmail,
     variables: {
       email: userEmail as string,
     },
   });
 
-  let initialFieldValues: RolesValues | undefined;
-  if (data?.projectById?.metadata) {
-    const metadata = data.projectById.metadata;
-    initialFieldValues = {
+  const { data: graphData } = useShaclGraphByUriQuery({
+    // do not fetch SHACL Graph until we get the project
+    // and the optional on chain credit class id associatied to it,
+    // this prevents from fetching this twice
+    skip: !project,
+    variables: {
+      uri: getProjectShapeIri(creditClassId),
+    },
+  });
```

Comment:
@haveanicedavid this is how we fetch the SHACL shape for project validation, see the comment in this query in particular